src/App.tsx

```diff
@@ -26,68 +27,106 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
+  switch (sortType) {
+    case SortType.ALPABET:
+      visibleGoods.sort((prev, current) => prev.localeCompare(current));
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((prev, current) => prev.length - current.length);
+      break;
+    case SortType.NONE:
+      goods.forEach((good, index) => {
+        visibleGoods[index] = good;
+      });
+      break;
+    default:
+
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleResetButton = (sortType !== SortType.NONE) || isReversed;
```

Comment:
```suggestion
  const isResetButtonVisible = (sortType !== SortType.NONE) || isReversed;
```
lets name boolean variables with is/are/has prefixes)